### PR TITLE
[interfaces/legacy] add Settings and make it accessible through xbmc.Addon().getSettings()

### PR DESCRIPTION
**Pull request by Montellese**
## Description
This PR adds a new `Settings` class to the Python interface which supports typed getters and setters for settings of an add-on. The add-on specific settings object can be retrieved using `xbmc.Addon().getSettings()`. 

## Motivation and context
Right now the main advantage of this settings wrapper object is that it also offers getters and setters for list settings by returning / expecting an `std::vector<>` instead of having to use `xbmc.Addon().setSetting()` with a stringified list of values. For my media import work I have already extended it to support some callbacks like for action settings when the user clicks on the setting or providing callbacks for dynamic setting values. Right now this is only available for media import add-ons but I plan to extend this to all python add-ons which will allow a much more interactive way of providing add-on settings.

## How has this been tested?
As part of my media import work.

## What is the effect on users?
Add-on developers will have access to a cleaner interface to interact with the settings of their add-ons.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [x] My change requires a change to the documentation, either Doxygen or wiki
- [x] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: xbmc/interfaces/legacy/Settings.h [hunk 1]
```diff
@@ -0,0 +1,502 @@
...
+/// @{
+/// @brief **Add-on settings**
+///
+/// \python_class{ Settings() }
+///
+/// This wrapper provides access to the settings specific to an add-on.
+/// It supports reading and writing specific setting values.
+///
+///-----------------------------------------------------------------------
+/// @python_v20 New class added.
```
Comment (enen92) [inline xbmc/interfaces/legacy/Settings.h:43]: Can you also add to methods please?

**Review by enen92 — APPROVED** [review]
Yet another one :) thanks a lot, this makes settings usage much cleaner from python

**Review by garbear — APPROVED** [review]
No mistakes that I can see

File: xbmc/interfaces/legacy/Settings.h (hunk 1 above)
Comment (Montellese) [reply xbmc/interfaces/legacy/Settings.h:43]: Sorry, added.

**Pull request merged**